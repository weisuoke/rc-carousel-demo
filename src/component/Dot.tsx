import * as React from 'react'
import * as styledComponents from 'styled-components'

export const Dot = styledComponents.default('span')<{ selected: boolean; selesctedColor?: string }>`
  display: inline-block
  width: 0.3rem
  height: 0.3rem
  border-radius: 0.15rem
  background-color: ${props => (props.selected ? props.selesctedColor || '#da373d' : 'grey')}
  margin: 0.3rem 0.15rem
  opacity: ${props => (props.selected ? 1 : 0.3)}
  transition-duration: 0.3s
`

export const WrapperDot = styledComponents.default('div')<{ width: number }>`
  position: absolute
  width: ${props => props.width}px
  z-index: 100
  bottom: 0.1rem
  text-align: center
  background-color: transparent
`

const IndicatorDot: React.SFC<{ total: number; currentIndex: number; width: number; selesctedColor: string }> = ({
  total,
  currentIndex,
  width,
  selesctedColor,
}) => {
  if (total === undefined || total < 4) {
    return <Dot selected={true} />
  } else {
    const len = total - 2
    const arr = Array.from(new Array(len).keys())
    return (
      <WrapperDot width={width}>
        {arr.map((dot, i) => {
          return <Dot selesctedColor={selesctedColor} selected={i === currentIndex - 1} key={i} />
        })}
      </WrapperDot>
    )
  }
}
export default IndicatorDot
