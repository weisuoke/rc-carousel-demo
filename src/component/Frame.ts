import * as styledComponents from 'styled-components'

export const Frame = styledComponents.default('div')<{ height: number }>`
  display: block
  width: 100vw
  height: ${props => props.height}px
  overflow: hidden
  background: grey
  position: relative
`
