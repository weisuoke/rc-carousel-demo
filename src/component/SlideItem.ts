import * as styledComponents from 'styled-components'

export const SlideItem = styledComponents.default('div')<{ width: number }>`
  display: inline-block
  height: 100%
  width: ${props => props.width}px
`
