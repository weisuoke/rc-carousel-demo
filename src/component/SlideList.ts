import * as styledComponents from 'styled-components'

interface SlideListProps {
  translateX: number
  width: number
}

export const SlideList = styledComponents.default.div.attrs<SlideListProps>({
  style: (props: SlideListProps) => ({
    width: props.width + 'px',
    transform: `translateX(${props.translateX}px)`,
  }),
})`
    display: block
    height: 100%
    position: relative
`
