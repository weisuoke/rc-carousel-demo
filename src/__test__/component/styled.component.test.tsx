import { configure, render } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import 'jest-styled-components'
import * as React from 'react'
import { Frame } from '../../component/Frame'
import { SlideItem } from '../../component/SlideItem'
import { SlideList } from '../../component/SlideList'

configure({ adapter: new Adapter() })

describe('it works', () => {
  it('Frame', () => {
    const wrapper = render(<Frame height={200} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('SlideList', () => {
    const slideListProps = {
      translateX: 200,
      width: 2000,
    }
    const wrapper = render(
      <SlideList {...slideListProps}>
        <SlideItem width={200} />
      </SlideList>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('SlideItem', () => {
    const wrapper = render(<SlideItem width={200} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
