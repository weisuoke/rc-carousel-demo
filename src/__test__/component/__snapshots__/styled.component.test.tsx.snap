// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works Frame 1`] = `
.c0 {
  display: block width:100vw height:200px overflow:hidden background:grey position:relative;
}

<div
  class="c0"
  height="200"
/>
`;

exports[`it works SlideItem 1`] = `
.c0 {
  display: inline-block height:100% width:200px;
}

<div
  class="c0"
  width="200"
/>
`;
